File: app/src/main/java/com/djrausch/billtracker/receivers/AlarmReceiver.java
package com.djrausch.billtracker.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.djrausch.billtracker.models.Bill;
import com.djrausch.billtracker.util.BillNotificationManager;
import com.djrausch.billtracker.util.BillUtil;

import io.realm.RealmResults;

/**
 * Created by white on 7/25/2016.
 */
public class AlarmReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        RealmResults<Bill> bills = BillUtil.loadOneWeekBillsForNotification();
        new BillNotificationManager(context).makeBillNotification(bills);
    }
}

File: app/src/main/java/com/djrausch/billtracker/receivers/AlarmReceiver.kt
package com.djrausch.billtracker.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.djrausch.billtracker.BillTrackerApplication
import com.djrausch.billtracker.models.Bill
import com.djrausch.billtracker.util.AlarmUtil
import com.djrausch.billtracker.util.BillUtil
import com.djrausch.billtracker.util.NotificationUtil

import org.joda.time.DateTime

import java.util.Date

import io.realm.RealmResults

/**
 * Created by white on 5/29/2016.
 */
class AlarmReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        //Find upcoming bills to show notificaiton for
        //Bill b = BillTrackerApplication.getRealm().where(Bill.class).findFirst();
        //RealmResults<Bill> bills = BillTrackerApplication.getRealm().where(Bill.class).findAll();
        //Find bills in the next week.
        val bills1 = BillUtil.loadOneWeekBillsForNotification()
        NotificationUtil.makeBillNotification(context, bills1)

    }
}
